### ISSUE TITLE
Does the mode with mqttmode 0 support TLS? 

### ISSUE THREAD

**Issue by coldy1992** [kubeedge/kubeedge#3728]
**What happened and what you expected to happen**:
I want the mqttmode 0 can support tls
**How to reproduce it (as minimally and precisely as possible)**:

**Anything else we need to know?**:

**Environment**:
- Kubernetes version (use `kubectl version`):
- KubeEdge version(e.g. `cloudcore --version` and `edgecore --version`):1.6.x
- Does the mode with mqttmode 0 support TLS？when the mqttmode is 0,I can't find an option to configure a certificate to start broker with tls

**Comment by fisherxu** [comment]
@coldy1992 Yes, now we only support Username/Password for mqtt in eventbus, and v1.6.1 is a old version... Do you want to use tls for the  mqtt in eventbus?

**Comment by coldy1992** [comment]
@fisherxu Yes,I want to use tls for the mqtt in eventbus.But I found that even the latest version of kubeedge, the mqttmode0 mode of eventbus does not support the use of TLS.Is that so？thank you

**Comment by fisherxu** [comment]
> @fisherxu Yes,I want to use tls for the mqtt in eventbus.But I found that even the latest version of kubeedge, the mqttmode0 mode of eventbus does not support the use of TLS.Is that so？thank you

Yes, because before we always run mqtt on localhost(127.0.0.1), so haven't add the tls. 
You will open the mqtt port? If need, would you like to open a pr to add it?

**Comment by coldy1992** [comment]
> 



> > @fisherxu Yes,I want to use tls for the mqtt in eventbus.But I found that even the latest version of kubeedge, the mqttmode0 mode of eventbus does not support the use of TLS.Is that so？thank you
> 
> Yes, because before we always run mqtt on localhost(127.0.0.1), so haven't add the tls. You will open the mqtt port? If need, would you like to open a pr to add it?

No, In my project I also run mqtt on localhost(127.0.0.1). I just want to confirm whether  mqttmode0 supports TLS.
OK,my question has been solved,Thank you very much!

**Comment by sdghchj** [comment]
MQTT server has two ways to authenticate a client. one by username/password, another by a client cert.
I see in v1.9 when external mqtt server is used, the mqtt client is able to set a client cert, however, whether surpprt tls transportation and client cert authentication depends on the external server side.
But, the internal mqtt server now listens without tls, so it doesn't surpport tls and client cert auth., and it use memory storage default with empty credentials, for security consideration, it is only for internal usage.

**Comment by coldy1992** [comment]
> MQTT server has two ways to authenticate a client. one by username/password, another by a client cert. I see in v1.9 when external mqtt server is used, the mqtt client is able to set a client cert, however, whether surpprt tls transportation and client cert authentication depends on the external server side. But, the internal mqtt server now listens without tls, so it doesn't surpport tls and client cert auth., and it use memory storage default with empty credentials, for security consideration, it is only for internal usage.

That's true, now I only need to use mqtt locally at the edge node, so... it's OK.Now I understand. Thank you so much！

**Comment by coldy1992** [comment]
> MQTT server has two ways to authenticate a client. one by username/password, another by a client cert. I see in v1.9 when external mqtt server is used, the mqtt client is able to set a client cert, however, whether surpprt tls transportation and client cert authentication depends on the external server side. But, the internal mqtt server now listens without tls, so it doesn't surpport tls and client cert auth., and it use memory storage default with empty credentials, for security consideration, it is only for internal usage.

That's true, now I only need to use mqtt locally at the edge node, so... it's OK.Now I understand. Thank you so much！
